contentreview-impl/impl/src/java/org/sakaiproject/turnitin/util/TurnitinLTIUtil.java

```diff
@@ -180,23 +184,46 @@ public int makeLTIcall(int type, String urlParam, Map<String, String> ltiProps){
 		return -5;
 	}
 	
-	public String getGlobalTurnitinLTIToolId(){
-		if(globalId == null){
-			log.debug("Setting global TII LTI tool id");
-			List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
-			if ( tools == null || tools.size() < 1  || tools.size() > 1) {
-				if(tools == null)
-					log.warn("No tools found");
-				else
-					log.warn("Found: " + tools.size());				
-				log.error("getGlobalTurnitinLTIToolId: wrong global TII LTI tool configuration");
-				return null;
+	public boolean obtainGlobalTurnitinLTIToolData(){
+		log.debug("Setting global TII LTI tool properties");
+		List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
```

Comment:
I had a look at it and didn't find a quick way to put it all just on one
place, there were circular dependencies. It doesn't have anything to do
with the actual jira anyway.

2016-06-03 15:56 GMT+02:00 Matthew Buckett <notifications@github.com>:

> In
> contentreview-impl/impl/src/java/org/sakaiproject/turnitin/util/TurnitinLTIUtil.java
> <https://github.com/ox-it/wl-turnitin/pull/17#discussion_r65710754>:
>
> > @@ -180,23 +184,46 @@ public int makeLTIcall(int type, String urlParam, Map<String, String> ltiProps){
> >  		return -5;
> >  	}
> >  	
> > -	public String getGlobalTurnitinLTIToolId(){
> > -		if(globalId == null){
> > -			log.debug("Setting global TII LTI tool id");
> > -			List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
> > -			if ( tools == null || tools.size() < 1  || tools.size() > 1) {
> > -				if(tools == null)
> > -					log.warn("No tools found");
> > -				else
> > -					log.warn("Found: " + tools.size());				
> > -				log.error("getGlobalTurnitinLTIToolId: wrong global TII LTI tool configuration");
> > -				return null;
> > +	public boolean obtainGlobalTurnitinLTIToolData(){
> > +		log.debug("Setting global TII LTI tool properties");
> > +		List<Map<String, Object>> tools = ltiService.getToolsDao("lti_tools.site_id = '"+turnitinSite+"'", null, 0, 0, turnitinSite);
>
> Were you going to fix this @bgarciaentornos
> <https://github.com/bgarciaentornos>?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ox-it/wl-turnitin/pull/17/files/25e0348519e93241319d2ede100c98570a883e31#r65710754>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AK6IluxK_9Rbd6N9QGjTMDGNFom0nidbks5qIDKHgaJpZM4H2DBM>
> .
>